### ISSUE TITLE
Keys do not expire because store::put() extends the key expire time

### ISSUE THREAD

**Issue by RatkoR** [GrahamCampbell/Laravel-Throttle#35]
Hi,

I have a middleware set to 100 hits per 2 minutes. If someone requests insane amount of api calls he'll get rate limited. All OK at this point.  

If normal users do an api requests (for example) once per 30 seconds, they'll get banned also. See why:  

When $throttle->attempt() checks if user is banned or not it calls hit() which increments the counter. [Code]
(https://github.com/GrahamCampbell/Laravel-Throttle/blob/master/src/Throttlers/CacheThrottler.php#L114)

The Store::put() calls MemcacheStore::put() and that one does memcached::set() [Code](https://github.com/illuminate/cache/blob/master/MemcachedStore.php#L61)

Here's the problem: Memcached::set() overwrites an object hence extending its time [lib info](http://docs.libmemcached.org/memcached_set.html?highlight=set#memcached_set). So we increment the counter and this counter is counting hits that happened long way back than 2 minutes. 

Example:
1st hit at 10:00:00 ... key value is 1, expire time is 10:02:00
2nd hit at 10.00.30 ... key value is 2, expire time is moved to 10:02:30 due to Memcaches:set()  
3rd hit at 10:01:00 ... key value is 3, expire time is 10:03:00
...
10th hit is at 10:05:00 ... key value is 10, expire time is again moved forward to 10:07:00 

Slowly, user will get banned, even thou he was under the rate limit set in middleware.

Any thought on this?

**Comment by GrahamCampbell** [comment]
Yeh, that was the intended behaviour, though I have considered changing it by storing a timestamp in with the cache so that we can manually flush it at our end.

**Comment by RatkoR** [comment]
Do you have a use case where this is an intended behaviour? I've been thinking about it and can't find one.  

It would also be nice if you'd add this the documentation. It'd help users if they suddenly start seeing wrong rate limits in action.

Changing:

$this->store->put($this->key, $this->number, $this->time); [here](https://github.com/GrahamCampbell/Laravel-Throttle/blob/master/src/Throttlers/CacheThrottler.php#L100)

to something like:
```
if ($this->count()) {
        $this->store->increment($this->key);
} else {
        $this->store->put($this->key, 1, $this->time);
}
```
discards key every two minutes (or in whatever number of minutes your middleware is set to) and starts counting from 0 again. People that do lots and lots of api requests will eventually get banned, and those that use apis normally wont.

**Comment by GrahamCampbell** [comment]
The use case was to prevent bots from attacking passwords.

**Comment by RatkoR** [comment]
Maybe, but even those requests (bots or not) would be wrongly rate limited, since I explicitly limited them at 100 in 2 minutes (not 100 in whatever minutes).

Anyway, thx for the throttle project. I'll just extend your CacheThrottler class and change pu() with increment() to limit users like I'd like to.

**Comment by GrahamCampbell** [comment]
NB, the correct way to extend this is NOT to fork my repo. The correct way is to require my repo as a dependency, then actually extend the class.

**Comment by GrahamCampbell** [comment]
That's why I wrote this code like this, with a factory class.

**Comment by GrahamCampbell** [comment]
This will be fixed in v5.0.0.
